extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceProvider.java

```diff
@@ -49,12 +49,25 @@ public KubernetesConfigSourceProvider(KubernetesConfigSourceConfig config, Kuber
                 if (log.isDebugEnabled()) {
                     log.debug("Attempting to read ConfigMap " + configMapName);
                 }
-                ConfigMap configMap = client.configMaps().withName(configMapName).get();
+                ConfigMap configMap = client.configMaps().inNamespace(config.namespace).withName(configMapName).get();
                 if (configMap == null) {
-                    logMissingOrFail(configMapName, client.getNamespace(), "ConfigMap", config.failOnMissingConfig);
+                    log.debug("ConfigMap " + configMapName + " does not exist in configured namespace " + config.namespace
+                            + ". Trying to read it from the current workspace " + client.getNamespace());
+                    configMap = client.configMaps().withName(configMapName).get();
```

Comment:
We probably don't want to do this, we would ideally just check if the Optional `namespace` property is set and act on way or the other

Reply:
Sorry about the delay. So if `namespace` property is set we search for the configmap in that namespace and `logMissingOrFail` if it is not founded, right?

Reply:
Yeah